contracts/base-extensions/BaseOptimisticRebalanceExtension.sol

```diff
@@ -0,0 +1,390 @@
+/*
+    Copyright 2023 Index Coop
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+pragma experimental "ABIEncoderV2";
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from  "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+import { AddressArrayUtils } from "../lib/AddressArrayUtils.sol";
+
+import { IAuctionRebalanceModuleV1 } from "../interfaces/IAuctionRebalanceModuleV1.sol";
+import { IBaseManager } from "../interfaces/IBaseManager.sol";
+import { ISetToken } from "../interfaces/ISetToken.sol";
+import { IDelegatedManager } from "../interfaces/IDelegatedManager.sol";
+import {BaseAuctionRebalanceExtension} from "./BaseAuctionRebalanceExtension.sol";
+import {AncillaryData } from "../lib/AncillaryData.sol";
+import { AssetAllowList } from "../lib/AssetAllowList.sol";
+import {OptimisticOracleV3Interface} from "../interfaces/OptimisticOracleV3Interface.sol";
+
+/**
+ * @title BaseOptimisticAuctionRebalanceExtension
+ * @author Index Coop
+ *
+ * @dev The contract extends `BaseAuctionRebalanceExtension` by adding an optimistic oracle mechanism for validating rules on the proposing and executing of rebalances. 
+ * It allows setting product-specific parameters for optimistic rebalancing and includes callback functions for resolved or disputed assertions.
+ */
+contract BaseOptimisticAuctionRebalanceExtension is  BaseAuctionRebalanceExtension, AssetAllowList {
```

Comment:
for consistency with [prior implementation](https://github.com/IndexCoop/index-coop-smart-contracts/blob/master/contracts/adapters/AuctionRebalanceExtension.sol), lets rename to `OptimisticAuctionRebalanceExtension`

Reply:
sure